### ISSUE TITLE
support: Malformed events when using NDKPrivateKeySigner

### ISSUE THREAD

**Issue by cypherhoodlum** [nostr-dev-kit/ndk#190]
Hi, I'm working on a Nostr project using NDK, and I have recently had to change from NIP-07 signers to NDKPrivateKeySigner.

Publishing with NIP-07 extensions worked just fine but now whenever I try to publish events signed with NDKPrivateSigner, I get the Error: No relay was able to receive the event.

I'm using versions:
```json
"@nostr-dev-kit/ndk": "2.1.1",
"@nostr-dev-kit/ndk-cache-dexie": "^2.1.1",
```

Here's a screenshot of a console.log of an event which was rejected by the relays

![Screenshot from 2024-03-15 18-59-51](https://github.com/nostr-dev-kit/ndk/assets/82510823/5fdd74cc-2d24-411b-b46c-abecd19c1389)

The way I use NDK is I create a singleton instance with this configuration: 

```javascript
this.dexieAdapter = new NDKCacheAdapterDexie({dbName: "eventsDB"})

this.signer = NDKPrivateKeySigner.generate()
// or in some cases
this.signer = new NDKPrivateKeySigner(hexPrivKey)

this.ndk = new NDK({
        signer: this.signer,
        explicitRelayUrls: this.explicitRelayUrls,
        enableOutboxModel: true,
        cacheAdapter: this.dexieAdapter,
        autoConnectUserRelays: false,
        client: "Nestr",
})
```

And this is how I have published events. To my understanding events don't need to be signed "manually" before publishing if the ndk signer is configured.

```javascript
// get the singleton class ndk instance
const ndk = ndkit.getNDK()

const publish = () => {
    const event = new NDKEvent(ndk)
    event.kind = 1
    event.content = noteContent
    event.ndk = ndk
    event.publish()
      .catch((error) => console.warn(`Note could not be published: ${error}`))
  }
```

I'm able to fetch events from relays so I know I'm connected to the relays properly. The only thing that has changed is the signer pretty much. Any ideas what might cause the problem? I appreciate the support.

**Comment by erskingardner** [comment]
Have you tried turning on Debugging with NDK using `localStorage.debug = 'ndk:*'` in your browser console? This might give you some more feedback about what is happening. The event in your screenshot looks fine to me and you're right, you shouldn't have to sign the event as a separate step, calling `publish` will also sign the event.

**Comment by cypherhoodlum** [comment]
TLDR; This issue can be closed. The problem was in the signer initialization logic. This was not a bug in NDK.

Thank you for responding so quickly. FYI the command `localStorage.debug = 'ndk:*'` doesn't seem to yield any logs in the Electron console, which is unfortunate since that's what we are using now in our project (hence the switch to NDKPrivateKeySigner). However, when I ran the app in a browser I got some error messages but they ended up being irrelevant.
 
I managed to fix it in the end. I think the problem was that I tried to re-initialize ndk with the new signer in too many places in the code after the user signs in with their private key. Not 100% what exactly caused the problem but it got solved by a bit of refactoring. Thank you for the support.

**Comment by erskingardner** [comment]
Cool. Glad you got it sorted, and thanks for letting me know.
